### ISSUE TITLE
Output of tracing-subscribe is verbose and difficult to read

### ISSUE THREAD

**Issue by alexcrichton** [bytecodealliance/wasmtime#7474]
I rand Wasmtime with `WASMTIME_LOG=wasmtime_wasi` just now and it looks like:

<img width="1530" alt="Screenshot 2023-11-03 at 10 57 13 AM" src="https://github.com/bytecodealliance/wasmtime/assets/64996/19e22be4-48a3-4227-b3ad-89250a802e8a">

Part of this is that my terminal screen isn't super wide but even on the widest setting

<img width="2560" alt="Screenshot 2023-11-03 at 10 57 33 AM" src="https://github.com/bytecodealliance/wasmtime/assets/64996/0dba0a84-4eb2-46b0-876c-aef5392e91ff">

I find this difficult to read.

I find that there's a ton of output per line. For example one log line looks like:

```
2023-11-03T15:56:46.567847Z \
  TRACE \
  wit-bindgen export{module="wasi:http/incoming-handler@0.2.0-rc-2023-11-05" function="handle"}:wit-bindgen \
  import{module="streams" function="[method]input-stream.read"}: \
  wasmtime_wasi::preview2::bindings::wasi::io::streams:
```

which takes up basically the entire screen before getting to the actual relevant message:

```
call self_=Resource { rep: 7, state: "borrow" } len=16384
```

Aesthetically I also find the highlighting quite confusing to read and sift through. I don't know if this has to do with my OS, terminal emulator, or what.

**Comment by pchickey** [comment]
Agreed we should clean this up. I vaugely recall the colors and highlighting looking a lot less terrible than this at one point, so there may be a knob we can tweak for that, but I haven't explored. The amount of context per line is far too much (by default anyway).
